projects/plugins/boost/app/assets/src/js/lib/utils/analytics.ts

```diff
@@ -53,18 +55,26 @@ function addBoostProps( props: TracksEventProperties ): TracksEventProperties {
 	const defaultProps: { [ key: string ]: string } = {};
 
 	/**
-	 * Jetpack Boost constant is not available on the front end.
+	 * jetpack_boost_ds constant is not available on the front end.
 	 *
 	 * So we need to check if it exists before using it in case this function is called from the front end.
 	 */
-	if ( typeof Jetpack_Boost !== 'undefined' ) {
-		if ( 'version' in Jetpack_Boost ) {
-			defaultProps.boost_version = Jetpack_Boost.version;
-		}
+	try {
+		defaultProps.boost_version = getConfig( 'version' );
+		const win = window as Window &
+			typeof globalThis & { jetpack_boost_ds?: { modules_state?: { value: ModulesState } } };
+
+		if ( win.jetpack_boost_ds?.modules_state?.value ) {
+			const value = win.jetpack_boost_ds.modules_state.value as ModulesState;
```

Comment:
As you mention in your PR comment, I'm not sure this is a good idea for a few reasons:

1. It reaches into the guts of DataSync and assumes data is there. If DataSync ever changes implementation, this will break.
2. It's doing it directly from inside a random function. If we were going to do something this messy, we should put it in a special function that just handles that bit of mess. But I don't think we should that even then.
3. Modules state can change during runtime. If this was an option like boost version number we can expect it to remain unchanged, but it will change during the runtime of the app. I really don't think we can rely on the window global.

That leaves us in an awkward spot for tracks events. I think we could make config a parameter we pass in whenever we track an event? That's a bit clunky, and it requires that you `useConfig` in whatever component triggers an event.

Maybe we need a `useAnalytics` which handles reading the config and returns a function you can call with tracks events details, and it handles integrating configuration info?